t/pubsub/kafka.t

```diff
@@ -0,0 +1,418 @@
+            local ok, err = ws:connect("ws://127.0.0.1:1984/kafka")
+            if not ok then
+                ngx.say("failed to connect: " .. err)
+                return
+            end
+
+            local data = {
+                {
+                    sequence = 0,
+                    cmd_kafka_list_offset = {
+                        topic = "not-exist",
+                        partition = 0,
+                        timestamp = -1,
+                    },
+                },
+                {
+                    sequence = 1,
+                    cmd_kafka_fetch = {
+                        topic = "not-exist",
+                        partition = 0,
+                        offset = 0,
+                    },
+                },
+                {
+                    sequence = 2,
+                    cmd_kafka_list_offset = {
+                        topic = "test-consumer",
+                        partition = 0,
+                        timestamp = -2,
+                    },
+                },
+                {
+                    sequence = 3,
+                    cmd_kafka_list_offset = {
+                        topic = "test-consumer",
+                        partition = 0,
+                        timestamp = -1,
+                    },
+                },
+                {
+                    sequence = 4,
+                    cmd_kafka_fetch = {
+                        topic = "test-consumer",
+                        partition = 0,
+                        offset = 14,
+                    },
+                }
+            }
+
+            for i = 1, #data do
+                local _, err = ws:send_binary(pb.encode("PubSubReq", data[i]))
+                local raw_data, raw_type, err = ws:recv_frame()
+                if not raw_data then
+                    ngx.say("failed to receive the frame: ", err)
+                    return
+                end
+                local data, err = pb.decode("PubSubResp", raw_data)
+                if not data then
+                    ngx.say("failed to decode the frame: ", err)
+                    return
+                end
+
+                if data.error_resp then
+                    ngx.say(data.sequence..data.error_resp.message)
+                end
+                if data.kafka_list_offset_resp then
+                    ngx.say(data.sequence.."offset: "..data.kafka_list_offset_resp.offset)
+                end
+                if data.kafka_fetch_resp then
+                    ngx.say(data.sequence.."offset: "..data.kafka_fetch_resp.messages[1].offset..
+                        " msg: "..data.kafka_fetch_resp.messages[1].value)
+                end
+            end
+
+            ws:send_close()
+        }
+    }
+--- response_body
+0failed to list offset, topic: not-exist, partition: 0, err: not found topic
+1failed to fetch message, topic: not-exist, partition: 0, err: not found topic
+2offset: 0
+3offset: 30
+4offset: 14 msg: testmsg15
```

Comment:
Let's add a comment about where the msg is from

Reply:
fixed #7032

Reply:
fixed #7032